# NanoComp


Compare multiple runs of Oxford Nanopore sequencing data and alignments

[![Twitter URL](https://img.shields.io/twitter/url/https/twitter.com/wouter_decoster.svg?style=social&label=Follow%20%40wouter_decoster)](https://twitter.com/wouter_decoster)
[![Build Status](https://travis-ci.org/wdecoster/nanocomp.svg?branch=master)](https://travis-ci.org/wdecoster/nanocomp)

### INSTALLATION
`pip install NanoComp`  

This script is written for Python3.

### USAGE
```
NanoComp [-h] [-v] [-t THREADS] [--readtype {1D,2D,1D2}] [-o OUTDIR]
                [-p PREFIX]
                [-f {eps,jpeg,jpg,pdf,pgf,png,ps,raw,rgba,svg,svgz,tif,tiff}]
                [-n [NAMES [NAMES ...]]]
                (--fastq [FASTQ [FASTQ ...]] | --summary [SUMMARY [SUMMARY ...]] | --bam [BAM [BAM ...]])


optional arguments:
  -h, --help            show this help message and exit
  -v, --version         Print version and exit.
  -t, --threads THREADS Set the allowed number of threads to be used by the script
  --readtype {1D,2D,1D2}
                        Which read type to extract information about from
                        summary. Options are 1D, 2D, 1D2
  -o, --outdir OUTDIR   Specify directory in which output has to be created.
  -p, --prefix PREFIX   Specify an optional prefix to be used for the output
                        files.
  -f, --format {eps,jpeg,jpg,pdf,pgf,png,ps,raw,rgba,svg,svgz,tif,tiff}
                        Specify the output format of the plots.
  -n, --names [NAMES [NAMES ...]]
                        Specify the names to be used for the datasets
  --plot {violin,box}   Which plot type to use: boxplot or violinplot (default)
  --fastq [FASTQ [FASTQ ...]]
                        Data is in default fastq format.
  --summary [SUMMARY [SUMMARY ...]]
                        Data is a summary file generated by albacore.
  --bam [BAM [BAM ...]]
                        Data as a sorted bam file.
```

### EXAMPLES
```
NanoComp --bam alignment1.bam alignment2.bam alignment3.bam --outdir compare-runs
NanoComp --fastq reads1.fastq.gz reads2.fastq.gz reads3.fastq.gz reads4.fastq.gz --names run1 run2 run3 run4
```


### EXAMPLE OUTPUT
![loglength example](https://github.com/wdecoster/nanocomp/blob/master/examples/NanoComp_log_length.png)
![box percentIdentity example](https://github.com/wdecoster/nanocomp/blob/master/examples/box_NanoComp_percentIdentity.png)

[See more examples](https://github.com/wdecoster/nanocomp/tree/master/examples)

I welcome all suggestions, bug reports, feature requests and contributions. Please leave an [issue](https://github.com/wdecoster/nanocomp/issues) or open a pull request. I will usually respond within a day, or rarely within a few days.
